server/src/utils/generate-verification-code.ts

```diff
@@ -0,0 +1,14 @@
+import crypto from "crypto";
+import moment from "moment";
+import config from "../../config.json";
+
+export default (number: string) => {
+  const today = moment().startOf("day");
+  const cipher = crypto.createHmac(
+    "sha256",
+    process.env.CIPHER_SECRET + today.toString()
+  );
+  cipher.update(number);
+  const result = cipher.digest("hex");
+  return result.slice(0, config.sms.codeLength);
```

Comment:
sms.codeLength is for mobile not email. Perhaps you will need a new arg on the function to switch this to email.codeLength.

Honenstly not sure about the config for this. Needs talking about.

Reply:
I agree. My suggestion is to pass the code length into the util. This has the advantage of making the util pure and therefore easier to test. We can than then call this util in a wrapper function that is specific to the the email / sms code generation.